articles/quickstart/native/android/03-session-handling.md

```diff
@@ -74,133 +108,67 @@ private final AuthCallback callback = new AuthCallback() {
 
     @Override
     public void onSuccess(@NonNull Credentials credentials) {
-        saveCredentials(credentials);
+        credentialsManager.saveCredentials(credentials);
         //...
     }
 };
 ```
 
 ::: note
-User credentials are stored in [Private mode](https://developer.android.com/reference/android/content/Context.html#MODE_PRIVATE) in the seed project in the `SharedPreferences` file. You can achieve this with the `CredentialsManager` class. For details, check the implementation in the project code. There are better and more secure ways to store tokens, but we will not cover them in this tutorial.
+The Storage implementation given to the Credentials Manager in the seed project uses a SharedPreferences file to store the user credentials in [Private mode](https://developer.android.com/reference/android/content/Context.html#MODE_PRIVATE). Change this behavior by implementing a custom Storage. 
 :::
 
-## Check for Tokens when the Application Starts
-
-To save your users the effort of logging in every time they open your app, use their access tokens. You can check for the access token when a user starts the app. If you find the token, you can automatically log the user in and direct them straight to your app's main flow.  
+## Recover the User's Credentials
 
+Retrieving the credentials from the Credentials Manager is an async process, as credentials may have expired and require to be refreshed. This renewing process is done automatically by the Credentials Manager as long as a valid refresh token is currently stored. A `CredentialsManagerException` exception will raise if the credentials have expired and the network request to refresh them has failed.
```

Comment:
~require to be refreshed.  This renewing process is done automatically~
require renewal. The renewal process is performed automatically

~A `CredentialsManagerException` exception will raise if the credentials have expired and the network request to refresh them has failed.~
A `CredentialsManagerException` exception will be raised if the credentials can not be renewed.